CollapseLauncher/Classes/RepairManagement/Zenless/ZenlessRepair.Fetch.cs

```diff
@@ -0,0 +1,467 @@
+
+            await foreach (FilePropertiesRemote asset in enumerable)
+            {
+                if (IsAudioFileIncluded(windowsFullPath, searchExclude, asset))
+                    yield return asset;
+            }
+        }
+
+        private static bool IsAudioFileIncluded(string windowsFullPath, SearchValues<string> searchInclude, FilePropertiesRemote asset)
+        {
+            // If it's not audio file, then return
+            ReadOnlySpan<char> relPath = asset.GetAssetRelativePath(out RepairAssetType assetType);
+            if (relPath.IsEmpty || assetType != RepairAssetType.Audio)
+                return true;
+
+            ReadOnlySpan<char> dirRelPath = Path.GetDirectoryName(relPath);
+            // If non language audio file, then return
+            if (dirRelPath.EndsWith(windowsFullPath))
+                return true;
+
+            // Check if non-full path, then return
+            int indexOf = dirRelPath.LastIndexOf(windowsFullPath);
+            if (indexOf < 0)
+                return true;
+
+            // If the index is more than WindowsFullPath.Length (included), then return
+            ReadOnlySpan<char> lastSequence = Path.GetFileName(dirRelPath);
+            int indexOfAny = lastSequence.IndexOfAny(searchInclude);
+
+            // Otherwise, return false
+            return indexOfAny == 0;
+        }
+        #endregion
+
+        #region Utilities
+#nullable enable
+        private async IAsyncEnumerable<FilePropertiesRemote> EnumerateResManifestToAssetIndexAsync(
+            IAsyncEnumerable<PkgVersionProperties> pkgVersion,
+            List<FilePropertiesRemote> assetIndex,
+            Dictionary<string, FilePropertiesRemote> hashSet,
+            string baseLocalPath,
+            string basePatchUrl,
+            string baseResUrl)
+        {
+            await foreach (FilePropertiesRemote? entry in pkgVersion.RegisterResCategorizedAssetsToHashSetAsync(assetIndex, hashSet, baseLocalPath, basePatchUrl, baseResUrl))
+            {
+                // If entry is null (means, an existing entry has been overwritten), then next
+                if (entry == null)
+                    continue;
+
+                yield return entry;
+            }
+        }
+#nullable restore
+
+        private void EliminatePluginAssetIndex(List<FilePropertiesRemote> assetIndex)
+        {
+            _gameVersionManager.GameAPIProp.data!.plugins?.ForEach(plugin =>
+            {
+                assetIndex.RemoveAll(asset =>
+                {
+                    return plugin.package!.validate?.Exists(validate => validate.path == asset.N) ?? false;
+                });
+            });
+        }
+
+        private string[] GetCurrentAudioLangList(string fallbackCurrentLangname)
+        {
+            // Initialize the variable.
+            string audioLangListPath = _gameAudioLangListPath;
```

Comment:
Any reason why the string needs to be copied in the method internal instead of using them directly?

Reply:
For debugging convenience. Sometimes it's hard for me to find which property actually responsible for the path.
Also, since the property's getter is checking the state every time it's getting its value, so we can just call it and set the value as in local instead of calling the getter multiple times.